### PR TITLE
Convert OpenTSDB to service

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This pull request converts the OpenTSDB server into a service and adds some tests.

### CODE REVIEW

File: services/opentsdb/handler.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	points := make([]tsdb.Point, 0, len(dps))
+	for i := range dps {
+		p := dps[i]
+
+		// Convert timestamp to Go time.
+		var ts time.Time
+		if p.Time < 10000000000 {
+			ts = time.Unix(p.Time, 0)
+		} else {
+			ts = time.Unix(p.Time/1000, (p.Time%1000)*1000)
```
Comment (corylanou) [inline services/opentsdb/handler.go]: I'm assuming we are converting to ms here?  But a comment to clarify would be helpful.

**Comment by corylanou** [comment]
+1 on green build.

**Comment by otoolep** [comment]
I also reviewed this code, particularly to see if it had implications for https://github.com/influxdb/influxdb/issues/2799. The WaitGroup  handling here seems OK, so +1 from me.

**Comment by otoolep** [comment]
Obviously +1 on a green build, as per @corylanou

**Comment by otoolep** [comment]
I don't where the PointsWriter is supplied to this service, by the production code.

File: services/opentsdb/handler.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+		}
+
+		points = append(points, tsdb.NewPoint(p.Metric, p.Tags, map[string]interface{}{"value": p.Value}, ts))
+	}
+
+	// Write points.
+	if err := h.PointsWriter.WritePoints(&cluster.WritePointsRequest{
+		Database:         h.Database,
+		RetentionPolicy:  h.RetentionPolicy,
+		ConsistencyLevel: cluster.ConsistencyLevelOne,
```
Comment (otoolep) [inline services/opentsdb/handler.go]: This looks hardcoded. Does this mean clients will have no control over consistency level with openTSDB?

Comment (benbjohnson) [inline services/opentsdb/handler.go]: I just want to convert the old server over to a service right now. We can add consistency level separately.

**Comment by benbjohnson** [comment]
@otoolep Ah, damn, good catch. Fixed in b688ecc.

**Comment by otoolep** [comment]
OK, let's get it in. Thanks for the final changes.

**Pull request merged**